src/wp-error-handling.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Set up basic error handling for WordPress.
+ *
+ * Initially we simply want to avoid error messages accidentally containing
+ * HTML in certain configurations. In the future we may improve the display
+ * beyond what PHP does by default.
+ *
+ * @package WordPress
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die();
+}
+
+/*
+ * The error_reporting() function can be disabled in php.ini. On systems where that is the case,
+ * it's best to add a dummy function to the wp-config.php file, but as this call to the function
+ * is run prior to wp-config.php loading, it is wrapped in a function_exists() check.
+ */
+if ( function_exists( 'error_reporting' ) ) {
+	/*
+	 * Initialize error reporting to a known set of levels.
+	 *
+	 * This will be adapted in wp_debug_mode() located in wp-includes/load.php based on WP_DEBUG.
+	 * @see https://www.php.net/manual/en/errorfunc.constants.php List of known error levels.
+	 */
+	error_reporting( E_CORE_ERROR | E_CORE_WARNING | E_COMPILE_ERROR | E_ERROR | E_WARNING | E_PARSE | E_USER_ERROR | E_USER_WARNING | E_RECOVERABLE_ERROR );
+}
+
+/**
+ * Handler for PHP's `set_error_handler`.
+ *
+ * We have two conflicting goals here:
+ *  1. We want to not print error text containing arbitrary HTML to the browser.
+ *  2. We want all PHP's built-in stuff like `error_get_last()` to work despite
+ *     bugs like https://bugs.php.net/bug.php?id=60575.
+ *
+ * Fortunately PHP doesn't normalize values set to `display_errors` via `ini_set()`, and
+ * it treats any unrecognized value as "off". So what we do here is save whatever is in
+ * `display_errors`, set it to "wp", print the error if appropriate, and then claims the
+ * error was unhandled so PHP does everything else it would normally do.
+ *
+ * @since 6.4.0
+ *
+ * @param int         $errno   Level of the error raised.
+ * @param string      $errstr  Error message.
+ * @param string|null $errfile Filename the error was raised in.
+ * @param int         $errline Line number where the error was raised.
+ *
+ * @return bool True to indicate the error was handled, false if not.
+ */
+function wp_error_handler( $errno, $errstr, $errfile, $errline ) {
+	static $display_errors = false;
+
+	$v = ini_get( 'display_errors' );
```

Comment:
I'd name this variable to something a bit more understandable. Maybe `$display_errors_ini_value`?

Reply:
Meh.

Reply:
Meh, or not, naming is important, because somebody will have to maintain the code you add to WordPress core. Please be more considerate and think about the code you write. It's read by humans, not by computers.